vispy/app/backends/_ipynb_rfb.py

```diff
@@ -0,0 +1,287 @@
+import asyncio
+
+from ..base import BaseApplicationBackend, BaseCanvasBackend, BaseTimerBackend
+from ...app import Timer
+from ...util import keys
+from ._offscreen_util import OffscreenContext, FrameBufferHelper
+
+
+try:
+    from jupyter_rfb import RemoteFrameBuffer
+except Exception:
+    RemoteFrameBuffer = object
+    _msg = 'ipynb_rfb backend relies on a the jupyter_rfb library: ``pip install jupyter_rfb``'
+    available, testable, why_not, which = False, False, _msg, None
+else:
+    available, testable, why_not = True, False, None
+    which = "jupyter_rfb"
+
+
+# -------------------------------------------------------------- capability ---
+
+capability = dict(
+    title=False,
+    size=True,
+    position=False,
+    show=False,
+    vsync=False,
+    resizable=True,
+    decorate=False,
+    fullscreen=False,
+    context=True,  # the offscreen global context
+    multi_window=True,
+    scroll=True,
+    parent=False,  # ipywidgets has layouts, but has no concept of parents
+    always_on_top=False,
+)
+
+
+# ------------------------------------------------------- set_configuration ---
+
+# The configuration mostly applies to the framebuffer. So if we'd want
+# to implement some of that, we'd probably have to apply it to the FBO.
+
+
+# ------------------------------------------------------------- application ---
+
+
+class ApplicationBackend(BaseApplicationBackend):
+
+    def __init__(self):
+        super().__init__()
+
+    def _vispy_get_backend_name(self):
+        return 'ipynb_rfb'
+
+    def _vispy_process_events(self):
+        raise RuntimeError("Cannot process events while asyncio event-loop is running.")
+
+    def _vispy_run(self):
+        pass  # We're in IPython; don't enter a mainloop or we'll block!
+
+    def _vispy_quit(self):
+        pass
+
+    def _vispy_get_native_app(self):
+        return asyncio
+
+
+# ------------------------------------------------------------------ canvas ---
+
+class CanvasBackend(BaseCanvasBackend, RemoteFrameBuffer):
+
+    _double_click_supported = True
+
+    def __init__(self, *args, **kwargs):
+        RemoteFrameBuffer.__init__(self)
+        BaseCanvasBackend.__init__(self, *args)
```

Comment:
Not that I necessarily have a problem with this, but couldn't you swap the order in the class definition and let `RemoteFrameBuffer`'s use of `super().__init__` handle this? Or maybe that's dumb to depend on that?

Reply:
Oh nice idea! I think I defined it in this order so that the Vispy-subclass appends to the chain of `ipywidget.DomWidget` classes, instead of fiddling it in at the top of the "mro-chain". But I'll have a look at least. 

I think generally you should be able to depend on each class to call `super().__init__()` but we break that rule in vispy (as found out the hard way) because for qt we call the `__init__` depending on qt4/qt5. Maybe I should try to clean that up so that each subclass can behave as expected.

Reply:
I'm looking into this, but I think I'll just clean this up properly, and replace the explicit `SomeBaseClass.__init__()` calls to `super().__init__()`, which will affect most backends. Let's do that in another PR.